### ISSUE TITLE
Proposal to work towards a better Taiko SR system.

### ISSUE THREAD

**Issue by mangomizer** [ppy/osu-performance#55]
With the recent qualification of HELIX (https://osu.ppy.sh/b/1584721&m=1), I wonder if it's finally time to have a look at updating/improving the SR algorithm in osu!taiko. 

**A bit of background:**
As far as I know, the SR algortihm is old, dating back to pre-2010. Back then the mapping meta was very different, following a more straightforward "authentic Taiko no Tatsujin style", and the SR algorithm was built in consideration of maps that were available at the time.* 

However, with increasing diversity in mapping styles and tastes, the past two years have seen a steady rise in the number of highly SR-inflated maps into osu!taiko's ranked section. As this is a competitive game, it is not surprising to see that these maps saturate the current meta as "must be played" in order for competitive players to maintain their ranking. From what I understand, it is a situation not too dissimilar from osu!. The SR algorithm is simply too dated to accurately judge the current meta, and an overhaul is long overdue.

**I should also add that apparently there was a council of taiko players that decided what was hard and what wasn't, and the story goes that it was decided that "doublets" were hard, and hence the reason why 1/8 doublets are so overweighted today.*

**What wrong with HELIX?**
There answer is that there is no problem with HELIX. HELIX [Inner Oni] is the latest and most extreme case of SR inflated maps - it is a 7.50* map that plays like a 5* map. The cause of this inflation is caused by the presence of many consecutive 1/8th doublets, which the game inaccurately deems to be extremely difficult. No one is disputing the map's timing/snaps, nor its spread progression - the mapset is excellent and plays perfectly fine. We should instead focus our energies on fixing the SR algorithm.

**What I propose we should do:**
I think the majority of the taiko playerbase already knows what is over/underrated with regards to osu!taiko's SR algorithm. I think the right way forward is to create an interactive tool that allows players to play around with the weighting of differnt components that contribute to the SR. In partiuclar, I would suggest addressing these 3 things as a priority:

1. abcd

1. 

Look into what makes 1/xth doublets overweighted, and nerf accordingly. Not all doublets are overweighted actually - look at this map for instance (https://osu.ppy.sh/b/1515547).

2. Look into 1/4th + 1/6th patterns, and nerf/buff accordingly.
3. Apply buff to raw speed (eg. >300bpm patterns)
4, TBC....

There are efforts already being made by certain community members in an attempt to recalculate SR (eg. https://osu.ppy.sh/forum/t/656048). From what I have seen, those attempts are an order of magnitude more accurate than what we currently have for the SR algorithm.

Given the success of osu! and osu!mania pp recalculations, I think it's about time we did something for osu!taiko too!

**TL;DR - Summary of fixes that I am proposing (in order of priority):**
Using an algorithm, 

Hopefully this post sheds some more light into the taiko SR situation, and maybe we can hope to see some real changes in the near future? Feel free to poke holes in my proposal/suggestion.

**Comment by Alchyr** [comment]
I made some changes to my thing.
https://osu.ppy.sh/forum/p/6311797#p6311797

**Comment by cbp105** [comment]
Yellow notes should be removed from SR calculation, or at least changed.  They can be ignored, but they still change SR.  Ranked maps don't have their SR changed too much by them, but there are loved maps at like 70* because of them.
